### PR TITLE
Propel2 Support – WIP

### PR DESCRIPTION
**Pull request by sebastianhoitz**
Just to reference these changes to other interested developers.

This is my first draft for Propel2 support. There are probably still a lot of cases where this is not enough. Especially the symfony propel1 bridge is still being used. But these changes allowed us to do CRUD operations on Propel2 models.

### CODE REVIEW

**Comment by K-Phoen** [comment]
Hey ! Thanks for taking the time to submit this PR :)

I'm aware that your work is still in a *WIP state* but I think that there is a major problem with your approach: it completely removes the ability to use Propel1 in this bundle.
Using Propel 1 or 2 should be as simple as configuring it. This probably means that for instance we'll have a `Datagrid/Propel1/ProxyQuery.php` and a `Datagrid/Propel2/ProxyQuery.php` (perhaps some code in common).

**Comment by Soullivaneuh** [comment]
> This probably means that for instance we'll have a Datagrid/Propel1/ProxyQuery.php and a Datagrid/Propel2/ProxyQuery.php (perhaps some code in common).

@K-Phoen This is completely BC break.

Can we find a more smooth approach?

If not, this should be on a new version (e.g. 2.0). About that, please see #62.

**Comment by K-Phoen** [comment]
Well, Propel2 being still in alpha we have probably some time to think about this :)

**Comment by Soullivaneuh** [comment]
@K-Phoen Yeah, but it's important to remember that! ;-)

**Comment by core23** [comment]
Are you still working on this? Please squash and rebase or close this PR @sebastianhoitz

File: Datagrid/Pager.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ public function init()
     protected function computeNbResults()
     {
         $nbResultsQuery = clone $this->getQuery();
-        $nbResultsQuery->limit(null)->offset(null);
+        //$nbResultsQuery->limit(null)->offset(null);
```
Comment (Soullivaneuh) [inline Datagrid/Pager.php]: No commented code before merge.

File: Model/ModelManager.php [hunk 2]
```diff
@@ -630,9 +637,10 @@ protected function getColumn($class, $property)
             return $this->cache[$class.'::'.$property] = $table->getColumn($property);
         }
 
-        if ($table && $table->hasColumnByInsensitiveCase($property)) {
-            return $this->cache[$class.'::'.$property] = $table->getColumnByInsensitiveCase($property);
-        }
+        // TODO Alternative?
```
Comment (Soullivaneuh) [inline Model/ModelManager.php]: No commented code before merge.

File: Datagrid/ProxyQuery.php [hunk 3]
```diff
@@ -13,7 +13,8 @@
 
 use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
 
-use ModelCriteria;
+//use ModelCriteria;
```
Comment (Soullivaneuh) [inline Datagrid/ProxyQuery.php]: No commented code before merge.

**Comment by sebastianhoitz** [comment]
Sorry, I'm not working on this anymore. So closing this PR.

**Pull request closed without merging**